apps/frontend/components/Escrow/LockFunds.tsx

```diff
@@ -46,39 +49,51 @@ const LockFunds = ({
 }) => {
   const chainId = useChainId();
   const { resolver, token, resolutionRate } = invoice;
-
+  const toast = useToast();
   const localForm = useForm();
   const { watch, handleSubmit } = localForm;
-
   const fee = formatUnits(
     resolutionRate === 0 ? BigInt(0) : BigInt(balance) / BigInt(resolutionRate),
     invoice.tokenMetadata.decimals
   );
   const feeDisplay = `${fee} ${parseTokenAddress(chainId, token)}`;
 
-  const disputeReason = useDebounce(watch('disputeReason'), 250);
-  const amount = formatUnits(BigInt(balance), invoice.tokenMetadata.decimals);
+  const disputeReason = useDebounce(watch('description'), 250);
+
+  const [txHash, setTxHash] = useState<Hex | undefined>(undefined);
 
   // const onSuccess = () => {
   //   // handle tx success
   //   // mark locked
   // };
 
-  const {
-    writeAsync: lockFunds,
-    writeLoading,
-    txHash,
-  } = useLock({
-    invoice,
-    disputeReason,
-    amount,
+  const queryClient = useQueryClient();
+
+  const onTxSuccess = () => {
+    queryClient.invalidateQueries({
+      queryKey: ['invoiceDetails'],
+    });
+  };
+
+  const { writeAsync: lockFunds, isLoading: writeLoading } = useLock({
+    invoice: { address: invoice.address, chainId },
+    localForm: localForm as unknown as UseFormReturn<FormLock>,
+    toast,
+    onTxSuccess,
+    details:
```

Comment:
why do we need to set details to zero bytes32 here? I think this then skips the ipfs upload inside `useLock`

Reply:
@dan13ram  https://github.com/raid-guild/dungeon-master-v2/pull/165/files/34e457f06c8128dace71b7a082ead72242e4cbba#diff-03bc35a4ab4e479a67a4d944e1b7f3cded94a29c39ae4d346dceffb6797bf203L27
it was like this in the removed useLock.ts so I moved it as an input to use it with useLock from smart invoice